### ISSUE TITLE
move away from sparkfun module to nRF52840 directly on PCB

### ISSUE THREAD

**Issue by tyalie** [tyalie/TyK380-keyboard#2]
This is a bit of a bigger overhaul, as it will mean that all the nRF52840 logic that is currently so neatly contained on the SparkFun module (e.g. antenna, clocks, flash) needs to be put directly on the PCB.

The very big advantage of this is, that we remove the SparkFun module and the M.2 connector. Both are very hard to source depending where you're from. Additionally the M.2 connector only works here because it has exactly the right height, which makes us prone to supply chain or deprecation issues. 

#### Tasks
- [x] prepare change by moving SparkFun module onto its own sheet
- [x] adapt public schematics for nRF module to our purposes
  - will require a bit of research but look for example to
    - https://gitlab.aachen.ccc.de/cpresser/catears
    - nRFMicro
- [ ] verify

**Comment by tyalie** [comment]
Okay so. I've managed to do most of the schematic. Only piece missing is the RF antenna.

**Comment by tyalie** [comment]
Okay. So I've aborted the idea of putting the nRF52 directly onto the board. The main problem is twofold:
1. it's really hard to get RF and MCU stuff right - especially the first time
2. it's essentially not possible to put traces everywhere without paying a pretty penny to the PCB manufacturer. The spacing between the pads on the nRF52840 package is far to small to really use all pins. For JLCPCB the extra cost would have been about 300% per PCB...

I've now decided on the HolyIoT board, which is available relatively cheaply, but is still very easy too use. Additionally I've moved to a gpio shift-register concept.
